packages/theming/apple-theme/src/createAppleTheme.macos.ts

```diff
@@ -2,15 +2,31 @@ import { ThemeReference } from '@fluentui-react-native/theme';
 import { Theme } from '@fluentui-react-native/theme-types';
 import { Appearance } from 'react-native';
 import { getBaseAppleThemeMacOS } from './appleTheme.macos';
+import { AccessibilityInfo } from 'react-native-macos';
+
+export let isHighContrastMode: boolean;
+let appleThemeReference: ThemeReference;
 
 export function createAppleTheme(): ThemeReference {
-  const appleThemeReference = new ThemeReference({} as Theme, () => {
+  appleThemeReference = new ThemeReference({} as Theme, () => {
     return getBaseAppleThemeMacOS();
   });
+  // Fetch initial system settings for high contrast mode
+  highContrastHandler();
+  // Invalidate theme and set prop when high contrast setting changes
+  AccessibilityInfo.addEventListener('highContrastChanged', () => {
+    highContrastHandler();
+  });
 
   Appearance.addChangeListener(() => {
     appleThemeReference.invalidate();
   });
-
   return appleThemeReference;
 }
+
+function highContrastHandler() {
+  AccessibilityInfo.isHighContrastEnabled().then((isEnabled) => {
+    appleThemeReference.invalidate();
+    isHighContrastMode = isEnabled;
```

Comment:
nit: I would set this before running invalidate, as the invalidate step depends on isHighContrastMode.